dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.MethodSource
+
+internal class DdAndroidGradlePluginFunctionalTest {
+    @TempDir
+    lateinit var testProjectDir: File
+    lateinit var mainSrcDir: File
+    lateinit var settingsFile: File
+    lateinit var buildFile: File
+    lateinit var manifestFile: File
+    lateinit var gradlePropertiesFile: File
+
+    @BeforeEach
+    fun `set up`() {
+        mainSrcDir = File(testProjectDir, "src/main").apply { mkdirs() }
+        settingsFile = File(testProjectDir, "settings.gradle")
+        buildFile = File(testProjectDir, "build.gradle")
+        gradlePropertiesFile = File(testProjectDir, "gradle.properties")
+        manifestFile = File(mainSrcDir, "AndroidManifest.xml")
+        stubFile(settingsFile, SETTINGS_FILE_CONTENT)
+        stubFile(manifestFile, ANDROID_MANIFEST_FILE_CONTENT)
+        stubFile(gradlePropertiesFile, GRADLE_PROPERTIES_FILE_CONTENT)
+    }
+
+    // region Assemble
+
+    @Test
+    fun `M success W assemble`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { build cache }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--build-cache", "assemble")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assembleDebug { configuration cache, checkProjectDependecies enabled }`() {
+        // The assembleRelease fails for configuration cache and not because of our
+        // code but because of these tasks: collect[Flavour1][Flavour2]ReleaseDependencies
+
+        // This test should fail because our `DdCheckSdkDepsTask` is not following the
+        // --configuration-cache best practices (we are using the project inside) and so far we
+        // could not easily solve that.
+
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("--configuration-cache", "assemble")
```

Comment:
shouldn't we test `assembleDebug` here per test name?

Reply:
Good catch.